src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapGet_NoParam_StringReturn_WithFilter.generated.txt

```diff
@@ -0,0 +1,206 @@
+
+                RequestDelegate targetDelegate = filteredInvocation is null ? RequestHandler : RequestHandlerFiltered;
+                var metadata = inferredMetadataResult?.EndpointMetadata ?? ReadOnlyCollection<object>.Empty;
+                return new RequestDelegateResult(targetDelegate, metadata);
+            }),
+
+        };
+
+        %GENERATEDCODEATTRIBUTE%
+        internal static Microsoft.AspNetCore.Builder.RouteHandlerBuilder MapCore<T>(
+            this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder routes,
+            string pattern,
+            System.Delegate handler,
+            IEnumerable<string> httpMethods,
+            string filePath,
+            int lineNumber)
+        {
+            var (populateMetadata, createRequestDelegate) = GenericThunks<T>.map[(filePath, lineNumber)];
+            return RouteHandlerServices.Map(routes, pattern, handler, httpMethods, populateMetadata, createRequestDelegate);
+        }
+
+        %GENERATEDCODEATTRIBUTE%
+        internal static Microsoft.AspNetCore.Builder.RouteHandlerBuilder MapCore(
+            this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder routes,
+            string pattern,
+            System.Delegate handler,
+            IEnumerable<string> httpMethods,
+            string filePath,
+            int lineNumber)
+        {
+            var (populateMetadata, createRequestDelegate) = map[(filePath, lineNumber)];
+            return RouteHandlerServices.Map(routes, pattern, handler, httpMethods, populateMetadata, createRequestDelegate);
+        }
+
+        internal static EndpointFilterDelegate BuildFilterDelegate(EndpointFilterDelegate filteredInvocation, EndpointBuilder builder, System.Reflection.MethodInfo mi)
+        {
+            var routeHandlerFilters =  builder.FilterFactories;
+            var context0 = new EndpointFilterFactoryContext
+            {
+                MethodInfo = mi,
+                ApplicationServices = builder.ApplicationServices,
+            };
+            var initialFilteredInvocation = filteredInvocation;
+            for (var i = routeHandlerFilters.Count - 1; i >= 0; i--)
+            {
+                var filterFactory = routeHandlerFilters[i];
+                filteredInvocation = filterFactory(context0, filteredInvocation);
+            }
+            return filteredInvocation;
+        }
+
+        internal static void PopulateMetadata<T>(System.Reflection.MethodInfo method, EndpointBuilder builder) where T : Microsoft.AspNetCore.Http.Metadata.IEndpointMetadataProvider
+        {
+            T.PopulateMetadata(method, builder);
+        }
+
+        internal static void PopulateMetadata<T>(System.Reflection.ParameterInfo parameter, EndpointBuilder builder) where T : Microsoft.AspNetCore.Http.Metadata.IEndpointParameterMetadataProvider
+        {
+            T.PopulateMetadata(parameter, builder);
+        }
+
+        internal static Task ExecuteObjectResult(object? obj, HttpContext httpContext)
```

Comment:
Do all of these methods need to be `internal`? Same for the `static class GenericThunks<T>` and the `static readonly Dictionary map`?

Reply:
Nope, the only things that need to be internal are the `MapCore` methods. The fact that these were internal was an artifact of the previous model where we used partial classes that separated out the `MapAction` definitions from the rest of the generated code.